api/server/router/local/image.go

```diff
@@ -184,12 +177,7 @@ func (s *router) postImagesPush(ctx context.Context, w http.ResponseWriter, r *h
 
 	name := vars["name"]
 	output := ioutils.NewWriteFlusher(w)
-	imagePushConfig := &graph.ImagePushConfig{
-		MetaHeaders: metaHeaders,
-		AuthConfig:  authConfig,
-		Tag:         r.Form.Get("tag"),
-		OutStream:   output,
-	}
+	imagePushConfig := graph.NewImagePushConfig(metaHeaders, authConfig, r.Form.Get("tag"), output)
```

Comment:
I'm not sure why we removing config structs everywhere. I thought we decided on like structs for >3 args. It's really hard to read such code.